### ISSUE TITLE
Query which beamlines are available

### ISSUE THREAD

**Issue by callumforrester** [DiamondLightSource/numtracker#10]
As a user I would like to be able to query which beamlines have configurations so that I can build queries in a discoverable way.

## Acceptance Criteria
- I can run a GraphQL query that tells me the names of beamlines with valid configurations for this service

**Comment by tpoliaw** [comment]
> As a user

As an admin?

**Comment by tpoliaw** [comment]
And RE `good first issue`, are you volunteering?

**Comment by tpoliaw** [comment]
With the current set of OPA rules, a regular user has no concept of 'their beamline'. It'd probably be possible to build a list of beamlines they've previously been on a visit on but it's questionable whether having been on  a visit a year ago should still allow them to query beamline info.

For the original ticket, there was an `all_beamlines` method removed in b234940 that could be reinstated to get part of the way there.

**Comment by callumforrester** [comment]
The other consideration is whether to make a new `configurations` query or simply expand the `configuration` one to return a list of configurations (where list size would be 1 for most use cases). The latter is simpler and probably more graphql-ish, since it allows for client-driven queries, the former aligns more the desire to separate "admin" queries from "user" queries. 

I'll probably expand the existing query unless you have strong feelings.

**Comment by tpoliaw** [comment]
That'd make sense. Make the `beamline` and optional input (renamed to filter?) and if absent return everything?

Or if you want to make the query more complicated but potentially more useful, have an optional field for each configuration option and only return the matching ones? It could be useful to be able to query for 'all beamlines that use template XYZ'. 

On the other hand, there's a decent argument for not adding anything until there's an actual use for it.

**Comment by callumforrester** [comment]
Yeah probably best not to overpredict. I think more complex queries can come from separate tickets. I'll just make a filter for the beamline for now.

On names we will eventually have to reconcile the names in this API with conventional names for entities in the supergraph (so off the top of my head we'll need to rename `beamline` to `instrument` in the API) but that can also be a separate ticket as-and-when.

**Comment by abbiemery** [comment]
From what I remember of the conversation we had about this last week I think we decided that `configuration` should take either list or None. Querying a given beamline would require `["beamline"]`, and you can query multiple `["beamline1", "beamline2"]`. If any of the beamlines do not have a configuration it should return a failed state. The returned list of configs could be a combination of valid and failed ones if you query something without one. Querying `None` would return all the beamlines that have a configuration.

Is this correct? @tpoliaw

**Comment by tpoliaw** [comment]
Sounds good to me assuming null counts as a 'failed state' in 

>If any of the beamlines do not have a configuration it should return a failed state
